### ISSUE TITLE
NG generate lazy loaded module

### ISSUE THREAD

**Issue by mbeckenbach** [angular/angular-cli#8717]
### Versions
```
Angular CLI: 1.5.5
Node: 6.9.5
OS: win32 x64
Angular: 5.0.4
```

### Feature request
Usually all of our feature modules are lazy loaded. 
So the developer has to run at least two commands to create a new module to get our default file layout. 

ng g m Foo -- routing
ng g c Foo --flat

Then the developer has to add a default route to the new modules routing module as well as the lazy loading route definition to the nearest parent module.

As the syntax and relative paths are allways the same, wouldn't it be possible to add some flags to combine those stepts?

maybe like this:
ng g module Foo --routing --component --lazy
or even simpler
ng g feature Foo (--lazy)

![image](https://user-images.githubusercontent.com/6654388/33491824-afb0cd74-d6bb-11e7-9d98-49542ffc9f53.png)

![image](https://user-images.githubusercontent.com/6654388/33491968-114e476e-d6bc-11e7-937c-7ee73a77f9e0.png)

**Comment by tomastrajan** [comment]
This can be prototyped with custom schematics eg: `ng generate lazy-feature` where lazy-feature will be a new schematic which can start as a part 3rd party collection and when stable enough can be included in main `@schematics/angular` repo?

**Comment by mbeckenbach** [comment]
@tomastrajan 
Sound very interesting. Is there any link to a guide on creating custom shcematics?

**Comment by tomastrajan** [comment]
@mbeckenbach check out this talk from Angular Connect 😉 

I am also interested in creating custom schematics but until something is done about this [issue](https://github.com/angular/devkit/issues/249) and this [PR](https://github.com/angular/angular-cli/pull/8723) you would have to include original `@schematics/angular` in your schematics and delegate calls for everything you are not implementing yourself. 

After the feature is merged it will be possible just to add your own schematics on top of default ones provided by Angular CLI just by adding your own collection in `angular-cli.json`.

Cheers!

**Comment by Robbie106gti** [comment]
They have made it fairly easy to do now, you can just do:
_

> _ng g m lazy --routing_

_

and add the route to your main router file/const

**Comment by doggy8088** [comment]
I think this feature has been implemented in [v8.1.0](https://github.com/angular/angular-cli/releases/tag/v8.1.0).

**Comment by clydin** [comment]
Closing as indicated above.
